src/comparator/field_comparator.py

```diff
@@ -1,63 +1,68 @@
+                self.field_update.label)
+            msg = 'Repeated state of the Field is changed, the original is {}, but the updated is {}'.format(
+                option_original, option_update)
+            FindingContainer.addFinding(
+                FindingCategory.FIELD_REPEATED_CHANGE, "", msg, True)
 
-        # 5. If the EnumDescriptors have the same repeated state, check if the type of them stay the same.
+        # 5. If the EnumDescriptors have the same repeated state,
+        # check if the type of them stay the same.
         if self.field_original.type != self.field_update.type:
-            type_original = FieldDescriptorProto().Type.Name(self.field_original.type)
-            type_update = FieldDescriptorProto().Type.Name(self.field_update.type)
-            msg = 'Type of the Field is changed, the original is {}, but the updated is {}'.format(type_original, type_update)
-            FindingContainer.addFinding(FindingCategory.FIELD_TYPE_CHANGE, "", msg, True)
-
-        # 6. Check the existing field is moved out of one-of or moved into one-of .
-        if self.field_original.containing_oneof != self.field_update.containing_oneof:
-            if self.field_original.containing_oneof != None:
-                msg = 'The Field {} is moved out of one-of'.format(self.field_original.name)
-                FindingContainer.addFinding(FindingCategory.FIELD_ONEOF_REMOVAL, "", msg, True)
-            else:
-                msg = 'The Field {} is moved into one-of'.format(self.field_update.name)
-                FindingContainer.addFinding(FindingCategory.FIELD_ONEOF_ADDITION, "", msg, True)
-                
-        # 7. Check `google.api.resource_reference` annotation.
-        # TODO(xiaozhenliu): annotation is removed, but the using file-level resource is added to the message.
-        # This should not be taken as breaking change.
-        # if (self.fieldDescriptor_original.options != None) and (self.fieldDescriptor_original.options['google.api.resource_reference'] != None):
-        #     print(pb2.DESCRIPTOR.message_types_by_name['ResourceReference'].fields[0])
-        #     print(FieldDescriptorProto.options.HasExtension)
-        #     print(descriptor_pb2.FieldOptions.DESCRIPTOR.message_types_by_name['ResourceReference'])
+            type_original = FieldDescriptorProto().Type.Name(
+                self.field_original.type)
+            type_update = FieldDescriptorProto().Type.Name(
+                self.field_update.type)
+            msg = 'Type of the field is changed, the original is {}, but the updated is {}'.format(
+                type_original, type_update)
+            FindingContainer.addFinding(
+                FindingCategory.FIELD_TYPE_CHANGE, "", msg, True)
 
+        # 6. Check `google.api.resource_reference` annotation.
+        # TODO(xiaozhenliu): annotation is removed, but the using
+        # file-level resource is added to the message.
+        # This should not be taken as breaking change.
```

Comment:
Really? Is there documentation for that somewhere?

Reply:
Sorry for the confusion, for my understanding from the design [doc](https://docs.google.com/document/d/16MrkP4C2Rp_wnFuZjV6uubm5Kt61BYcPQwfyFMugOD8/edit#) If the `google.api.resource_reference` annotation is removed, it is a breaking change unless the reference was to itself, using a file-level resource annotation, and the file-level annotation has been moved into this message.

```
option (google.api.resource_definition) = {
  type: "iam.googleapis.com/ServiceAccount"
  pattern: "projects/{project}/serviceAccounts/{service_account}"
};

message GenerateAccessTokenRequest {
  string name = 1 [
    (google.api.field_behavior) = REQUIRED,
    (google.api.resource_reference).type = "iam.googleapis.com/ServiceAccount",
  ];
  .....
}
// .... nowhere else uses ServiceAccount resource definition.
```
This should be equivalent to 
```
// file-level resource definition is removed.
message GenerateAccessTokenRequest {
  option (google.api.resource) = {
    type: "iam.googleapis.com/ServiceAccount",
    pattern: "projects/{project}/serviceAccounts/{service_account}"
  };
  string name = 1 [
    (google.api.field_behavior) = REQUIRED,
    (google.api.resource_reference).type = "iam.googleapis.com/ServiceAccount",
  ];
  .....
}
```

Reply:
Yes, I believe this is correct, and is as specified in the doc.